pjrt/convert/trace_container.zig

```diff
@@ -0,0 +1,211 @@
+            var line = xplane_visitor.XLineVisitor.init(plane, l);
+            const resource_id: u32 = @intCast(line.displayId());
+            var resource: trace_events_proto.Resource = .{
+                .resource_id = resource_id,
+                .name = .{ .Const = line.displayName() },
+            };
+
+            if (sort_by_ordinal) {
+                ordinal += 1;
+                resource.sort_index = ordinal;
+            }
+            try device.resources.put(allocator, resource_id, resource);
+        }
+    }
+
+    fn xplaneToTraceEvents(self: *TraceContainer, allocator: std.mem.Allocator, device_id: u32, xplane: *const xplane_visitor.XPlaneVisitor) !void {
+        // Convert devices and resources.
+        const device_entry = try self.metadata.devices.getOrPut(allocator, device_id);
+        if (!device_entry.found_existing) device_entry.value_ptr.* = .{};
+
+        try buildDeviceAndResources(allocator, device_id, xplane, device_entry.value_ptr);
+
+        // Convert events.
+        for (xplane.plane.lines.items) |*l| {
+            const xline = xplane_visitor.XLineVisitor.init(xplane, l);
+            const resource_id: u32 = @intCast(xline.displayId());
+            if (std.mem.eql(u8, xline.displayName(), xla_async_op_line_name)) continue;
+            for (xline.line.events.items) |*e| {
+                const xevent = xplane_visitor.XEventVisitor.init(xline.plane, xline.line, e);
+                const event_type: xplane_schema.HostEventType = xevent
+                    .type_ orelse .UnknownHostEventType;
+                if (event_type.isInternalEvent()) continue;
+                var event = try self.createEvent(allocator);
+                event.device_id = device_id;
+                event.resource_id = resource_id;
+                if (xevent.displayName()) |name| {
+                    event.name = name;
+                    try event.args.put(allocator, "long_name", xevent.name());
+                } else {
+                    event.name = xevent.name();
+                }
+                event.timestamp_ps = @intCast(xevent.timestampPs());
+                event.duration_ps = @intCast(xevent.durationPs());
+
+                const xevent_md_visitor = xevent.metadataVisitor();
+                for (xevent_md_visitor.stats_owner.stats.items) |*s| {
+                    const xstat = xplane_visitor.XStatVisitor.init(xevent_md_visitor.plane, s);
+                    if (xstat.stat.value == null) continue;
+                    const stat_str = try xstat.toString(allocator);
+                    if (xstat.type()) |t| {
+                        if (t.isInternalStat()) continue;
+                        if (t == .step_name) {
+                            event.name = stat_str;
+                        }
+                    }
+                    try event.args.put(allocator, xstat.name(), stat_str);
+                }
+
+                for (xevent.event.stats.items) |*s| {
```

Comment:
here also there is obfuscation: xevent.event is just our original `e`. I don't think this "renaming"  helps